From 8f848593293b69f293734e07ec975ee76a3e6df5 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Sun, 19 Jun 2022 10:39:12 +0200
Subject: [PATCH] test: skip TEST-21-DFUZZER without ASan

as the test is quite time consuming and it yields more useful reports
when the target app is built with sanitizers.

(cherry picked from commit d768243a95c33e73afe9a7e487acf329884e03c1)

Related: #2087652
---
 test/TEST-21-DFUZZER/test.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/TEST-21-DFUZZER/test.sh b/test/TEST-21-DFUZZER/test.sh
index ecc04e368c..42e37c8a9c 100755
--- a/test/TEST-21-DFUZZER/test.sh
+++ b/test/TEST-21-DFUZZER/test.sh
@@ -12,6 +12,11 @@ QEMU_TIMEOUT="${QEMU_TIMEOUT:-1800}"
 
 command -v dfuzzer >/dev/null || exit 0
 
+if ! get_bool "$IS_BUILT_WITH_ASAN"; then
+    echo "systemd is built without ASan, skipping..."
+    exit 0
+fi
+
 test_append_files() {
     local workspace="${1:?}"
 
